src/main/java/com/hms_networks/americas/sc/extensions/historicaldata/RapidCatchUp.java

```diff
@@ -0,0 +1,223 @@
+package com.hms_networks.americas.sc.extensions.historicaldata;
+
+import com.ewon.ewonitf.Exporter;
+import com.hms_networks.americas.sc.extensions.datapoint.DataPoint;
+import com.hms_networks.americas.sc.extensions.json.JSONException;
+import com.hms_networks.americas.sc.extensions.string.StringUtils;
+import com.hms_networks.americas.sc.extensions.system.time.SCTimeUtils;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.util.List;
+
+/**
+ * This class addresses the need for historical data tracking to "catch up" quickly when there are
+ * periods where the tracker is not running. For example, the Flexy is powered off for several
+ * weeks. For correct behavior, it is required that injectJvmLocalTime from {@link SCTimeUtils} is
+ * called at startup.
+ *
+ * @author HMS Networks, MU Americas Solution Center
+ * @since 1.14.2
+ */
+public class RapidCatchUp {
+
```

Comment:
```suggestion

  /** The number of milliseconds in one second. */
  private static final long MILLISECONDS_PER_SECOND = 1000L;

   /** The number of milliseconds in one minute. */
  private static final long MILLISECONDS_PER_MINUTE = 60000L;

   /** The number of milliseconds in one hour. */
  private static final long MILLISECONDS_PER_HOUR = 3600000L;
```

Reply:
These constants exist elsewhere in the extensions lib (src\main\java\com\hms_networks\americas\sc\extensions\system\time\SCTimeUnit.java) and could probably just be made public since they are really useful when dealing with any time related functions on the Flexy

Reply:
Will make constants public.